lib/plugins/aws/rollback.js

```diff
@@ -77,7 +77,7 @@ class AwsRollback {
       throw new ServerlessError(`${msg} ${hint}`, 'ROLLBACK_DEPLOYMENT_NOT_FOUND');
     }
 
-    service.package.artifactDirectoryName = `${prefix}/${dateString}`;
+    service.provider.s3DeploymentDirectoryPath = `${prefix}/${dateString}`;
```

Comment:
I think here what should be overriden is just `serverless.getProvider('aws').deploymentTimestamp` and having that getters should start resolve the right deployment path (?)